src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -263,7 +263,7 @@ private void registerExtendedClass(String qualifiedClassName,
         final String superClassName = getSuperClassName(currentClass.getTypeDeclarationAst());
         if (superClassName != null) {
             final ToIntFunction<ClassDesc> nestedClassCountProvider = classDesc -> {
-                return CheckUtil.typeDeclarationNameMatchingCount(qualifiedClassName,
+                return typeDeclarationNameMatchingCount(qualifiedClassName,
```

Comment:
Please move this change to separate PR, let's not make this PR heavy  in changes to let us review and merge quickly.